src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -92,13 +94,16 @@
       File="$(AssemblyInfoFile)"
       Lines="@(AssemblyInfoUsings);@(AssemblyInfoLines)"
       Overwrite="true" />
-
-    <ItemGroup>
-      <Compile Include="$(AssemblyInfoFile)" />
-      <FileWrites Include="$(AssemblyInfoFile)" />
-    </ItemGroup>
   </Target>
 
+  <ItemGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
+    <Compile Include="$(AssemblyInfoFile)">
```

Comment:
For some solutions, such as Microsoft.CSharp, _AssemblyInfo.cs is generated when the solution is opened, if it hadn't been generated previously.  For others, such as System.IO.sln, the link is broken until the project is built.